CRAYSAT-1549: Document SAT inclusion in CSM (#23)
Cray-HPE/docs-sat: docs/introduction.md

- [Grafana Fabric Port State Dashboard](dashboards/SAT_Grafana_Dashboards.md#grafana-fabric-port-state-dashboard)
- [Grafana Fabric RFC3635 Dashboard](dashboards/SAT_Grafana_Dashboards.md#grafana-fabric-rfc3635-dashboard)

> In CSM 1.3 and newer, the `sat` command is automatically available on all the Kubernetes NCNs. See

Author: Correct, when it comes to the NCN images there is no granularity between the management and worker NCNs. Our documentation should continue to state that SAT is only supported on the manager NCNs, but this is a potential inconsistency.

[SAT in CSM](#sat-in-csm) for more information. Older versions of CSM do not have the `sat` command automatically

> available, and SAT must be installed as a separate product as part of the HPE Cray EX System base installation.

Reviewer: It was part of the original text, but I was wondering this as well. Would it be better to just remove that statement if it doesn't have any value?

> available, and SAT must be installed as a separate product as part of the HPE Cray EX System base installation.

Author: I'm going to plan on removing it. It must have been referring to the CSM product as the "base installation" or possibly the node image, but I think "separate product" is clear enough on its own given how the Cray EX software is structured into "products".

> available, and SAT must be installed as a separate product as part of the HPE Cray EX System base installation.

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

(CONTAINER_ID) sat-container# sat status
```

> ## SAT in CSM 1.3

Reviewer: I'd suggest changing the title of this section to the following. In a few years, most customers will be on CSM 1.3 or later, and we won't need to call out the version in this section title. However, if we change the title then, I believe all links to it in other EX documents will break. I think it's ok to change this now since we list the specific CSM version in the first line of text.
```suggestion
## SAT in CSM
```

> 

Reviewer: Line 29 of this file is incorrect for systems running CSM 1.3. I'd suggest updating it to something like the following:

In CSM 1.3 and newer, the `sat` command is automatically available on all the Kubernetes NCNs. See [SAT in CSM] (introduction.md#sat-in-csm) for more information. Older versions of CSM do not have the `sat` command automatically available, and SAT must be installed as a separate product as part of the HPE Cray EX System base installation

Author: Thanks, addressed in 9609bb3c912e676d49f5569bee883d6c1dd33634.

(CONTAINER_ID) sat-container# sat status
```

## SAT in CSM

In CSM 1.3 and newer, the `sat` command is automatically available on all the Kubernetes NCNs, though it is still
possible to install SAT as an independent product stream. Installing the SAT product stream results in additional

> supporting components being installed.

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

- Installing the SAT product stream creates an entry for SAT in the `cray-product-catalog` Kubernetes ConfigMap.
  Without installing the SAT product stream, there will be no entry for this version of SAT in the output of

>   `sat showrev`.

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

> 

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

- The `sat-install-utility` container image is only available with the full SAT product stream. This container image
  provides uninstall and activate functionality when used with the `prodmgr` command. It is still possible to uninstall

>   previous versions of the SAT product stream.

Author: Just to respond to this one quickly - the reworded version is maybe a little misleading. It's confusing but the most correct, detailed thing to say might be something like 'It is still possible to uninstall previous versions of the SAT product stream because those versions come packaged with `sat-install-utility`.

In other words, a version of `sat-install-utility` is packaged with each product stream release and is used for uninstalling that version of the product stream.

>   previous versions of the SAT product stream.

Reviewer: Are the previous versions of SAT you're referring to those before SAT was available in CSM 1.3 (so SAT 2.3 and older)? Is this a distinction we should make if SAT 2.4 and newer can't be uninstalled unless the full SAT product stream is installed on an EX system? Maybe we add something like the following as a new line indented underneath this bullet:

NOTE: In SAT 2.3 and older, SAT was only available to install as a separate product stream. Because these versions were packaged with `sat-install-utility`, it is still possible to uninstall these versions of SAT.

>   previous versions of the SAT product stream.

Author: Yeah, that is pretty much right.

Although to add to the confusion, 2.4 is really a version of the product stream, so it doesn't exactly make sense to say "SAT 2.4 is in CSM 1.3", you might say something like "the version of the SAT CLI that is in SAT 2.4 is also in CSM 1.3" or something.

I like the note there, I think that's totally accurate to say. Maybe we could condense it all and say,
```
- The `sat-install-utility` container image is only available with the full SAT product stream. This container image
  provides uninstall and activate functionality when used with the `prodmgr` command. In SAT 2.3 and older, SAT was
  only available to install as a separate product stream. Because these versions were packaged with
  `sat-install-utility`, it is still possible to uninstall these versions of SAT.
```

I'd be open to some more feedback here, if you want to further refine that.

>   previous versions of the SAT product stream.

Reviewer: I'm ok with including the note with the bullet if we put parenthesis around it. I'd like to somehow distinguish visually that this is additional information for older SAT versions. That way it doesn't disrupt too much the flow of the bulleted list overall. What do you think of the following?

- The `sat-install-utility` container image is only available with the full SAT product stream. This container image
  provides uninstall and activate functionality when used with the `prodmgr` command. (In SAT 2.3 and older, SAT was
  only available to install as a separate product stream. Because these versions were packaged with
  `sat-install-utility`, it is still possible to uninstall these versions of SAT.)

>   previous versions of the SAT product stream.

Author: Yeah, that looks awesome, thanks!

>   previous versions of the SAT product stream.

Author: I implemented the suggestion in 45fae7c23fd959bdca55482f63824e0422d6509c

- The `sat-config-management` git repository in Gitea (VCS) and thus the SAT layer of NCN CFS configuration is

>   only available with the full SAT product stream. See [SAT CFS Configuration](#sat-cfs-configuration-details) below.

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

- The `docs-sat` RPM package is only available with the full SAT product stream.

### SAT CFS Configuration Details

> When the SAT product stream has not been installed, there will be no configuration content for SAT in VCS. Therefore,

Reviewer: I have a few suggestions to help the flow of the rest of this information:
```suggestion
If the SAT product stream is not installed, there will be no configuration content for SAT in VCS. Therefore,
```

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

CFS configurations that apply to NCNs (for example, `ncn-personalization`) should not include a SAT layer.

> This configuration layer modifies the permissions of files left over from prior installs of SAT, so that the Keycloak
> username used to authenticate to the API gateway cannot be read by users other than `root`. Specifically, it does the

Reviewer: ```suggestion
The SAT configuration layer modifies the permissions of files left over from prior installations of SAT, so that the Keycloak
username that authenticates to the API gateway cannot be read by users other than `root`. Specifically, it does the
```

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

following:

> - Modify the `sat.toml` configuration file which contains the username so that it is only readable by `root`.

Reviewer: ```suggestion
- Modifies the `sat.toml` configuration file which contains the username so that it is only readable by `root`.
```

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

> - Modify the `/root/.config/sat/tokens` directory so that the directory is only readable by `root`, because the
>   names of the files within the `tokens` directory contain the username.

Reviewer: ```suggestion
- Modifies the `/root/.config/sat/tokens` directory so that the directory is only readable by `root`. This is needed because the
  names of the files within the `tokens` directory contain the username.
```

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c

> Regardless of this configuration being applied, the contents of the tokens are never readable by other users, nor are
> any passwords. These permission changes only apply to files created by previous installs of SAT; in the current version
> of SAT all files and directories are created with the appropriate permissions.

Reviewer: ```suggestion
Regardless of the SAT configuration being applied, passwords and the contents of the tokens are never readable by other users. 
These permission changes only apply to files created by previous installations of SAT. In the current version
of SAT, all files and directories are created with the appropriate permissions.
```

Author: Addressed in 45fae7c23fd959bdca55482f63824e0422d6509c
